internal/webhookhandler/webhookhandler.go

```diff
@@ -78,19 +78,27 @@ func (p *podSidecarInjector) Handle(ctx context.Context, req admission.Request)
 	ns := corev1.Namespace{}
 	err = p.client.Get(ctx, types.NamespacedName{Name: req.Namespace, Namespace: ""}, &ns)
 	if err != nil {
-		return admission.Errored(http.StatusInternalServerError, err)
+		res := admission.Errored(http.StatusInternalServerError, err)
+		// By default, admission.Errored sets Allowed to false which blocks pod creation even though the failurePolicy=ignore.
+		// Allowed set to true makes sure failure does not block pod creation in case of an error.
+		res.Allowed = true
+		return res
 	}
 
 	for _, m := range p.podMutators {
 		pod, err = m.Mutate(ctx, ns, pod)
 		if err != nil {
-			return admission.Errored(http.StatusInternalServerError, err)
+			res := admission.Errored(http.StatusInternalServerError, err)
+			res.Allowed = true
+			return res
```

Comment:
I am not sure whether creating an error and later overriding `allowed` is the best option - or what happens in the admission controller when `allowed = true` but `result.code != 200`.

I would have suggested to create an allowed response but add warnings to it like so:

```suggestion
			return admission.Allowed("").WithWarnings(err.Error())
```

Having said that, I am fairly new to the workings of an admission controller and admission webhooks, so maybe your solution is already best practice.

Reply:
I will play around and try to add events explicitly to the pod.

Reply:
Something like this

```
p.recorder.Event(&pod, "Normal", "ConfigUpdate", fmt.Sprintf("Testing events"))
```

does not work in the webhook because the pod name is not known. It creates following event:

```
5s          Normal    ConfigUpdate        pod                                                Testing events
```

Reply:
Based on ^^ I am keeping my original version of this PR and I have added more comments with findings to the code.